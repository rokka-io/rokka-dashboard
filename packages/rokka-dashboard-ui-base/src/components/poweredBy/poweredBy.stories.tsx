import { storiesOf } from '@storybook/react';
import React from 'react';
import { colors } from '../../identity/colors/colors';
import { PoweredBy } from './poweredBy';

storiesOf('Components / Sidebar', module)
  .addDecorator(story => (
    <div style={{ width: '100vw', height: '64px', padding: '10px', background: colors.gray.sidebar, position: 'relative' }}>
      {story()}
    </div>
  ))
  .add('PoweredBy', () => <PoweredBy />);
