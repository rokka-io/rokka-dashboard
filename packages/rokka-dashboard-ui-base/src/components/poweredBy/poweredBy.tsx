import React from 'react';
import styled from 'styled-components';
import { LiipIcon } from '../../elements';
import { colors } from '../../identity/colors/colors';
import { spaces } from '../../identity/spaces/spaces';
import { fonts } from '../../identity/typography';

export function PoweredBy() {
  return (
    <StyledLink href="https://www.liip.ch" target="_blank" rel="noopener">
      POWERED BY <StyledLiipIcon height="16px" />
    </StyledLink>
  );
}

const StyledLink = styled.a`
  transition: 0.4s ease opacity;
  color: ${colors.tints.white};
  font-size: ${fonts.Sizes.large};
  text-decoration: none;
  position: absolute;
  bottom: ${spaces.medium};
  left: ${spaces.medium};
  opacity: 0.2;

  :hover {
    transition: 0.4s ease opacity;
    opacity: 1;
  }
`;

const StyledLiipIcon = styled(LiipIcon)`
  display: inline-block;
  margin-top: -4px;
  margin-left: 2px;
`;
