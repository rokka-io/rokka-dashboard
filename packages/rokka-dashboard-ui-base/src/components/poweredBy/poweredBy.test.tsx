import { shallow } from 'enzyme';
import React from 'react';
import { PoweredBy } from './poweredBy';

describe('Components', () => {
  describe('Powered By', () => {
    it('renders a Powered By component.', () => {
      const component = shallow(<PoweredBy />);

      expect(component).toMatchSnapshot();
    });
  });
});
