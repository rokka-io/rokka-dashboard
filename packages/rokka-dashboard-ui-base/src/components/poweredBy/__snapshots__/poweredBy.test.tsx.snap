// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components Powered By renders a Powered By component. 1`] = `
<styled.a
  href="https://www.liip.ch"
  rel="noopener"
  target="_blank"
>
  POWERED BY 
  <Styled(LiipIcon)
    height="16px"
  />
</styled.a>
`;
